OpenRA.Mods.Common/Traits/Player/ParallelProductionQueue.cs

```diff
@@ -50,6 +50,11 @@ public override bool IsProducing(ProductionItem item)
 			return Queue.Contains(item);
 		}
 
+		protected override void BeginProduction(ProductionItem item, bool queued)
+		{
+			base.BeginProduction(item, true);
```

Comment:
Why don't you pass in the queued parameter here?

Reply:
Should I add an inline comment there?

Reply:
It would be better to more clearly replace the behavior instead of manipulating the arguments - just replace this with the `Queue.Add(item);` instead.

Reply:
:+1: to pchotes comment

Reply:
I've reworked the method as suggested.

Reply:
Sorry, didnt find time to review this, now its merged, but the base.BeginProduction was madatory. while it makes no difference for the ParallelProductionQueue, this basicaly overrides any custom code in the base method, which my InfiniteBuildAfter PR adds :-P ill just revert this in my other PR then.

Reply:
Yeah, I thought about that too after looking at your PR #15021